ext/standard/math.c

```diff
@@ -812,7 +813,7 @@ PHPAPI void _php_math_basetozval(zend_string *str, int base, zval *ret)
 	}
 
 	if (invalidchars > 0) {
-		zend_error(E_DEPRECATED, "Invalid characters passed for attempted conversion, these have been ignored");
```

Comment:
I would suggest using `zend_value_error("...")`. This is what we usually use lately in case of problems with the supplied data itself.

Reply:
Indeed this makes way more sense and doesn't reply on SPL (which is a bit of a mess with in it self)